### ISSUE TITLE
Search input field: design option to delete input

### ISSUE THREAD

**Issue by inesdgomes** [international-labour-organization/designsystem#708]
⚠️ This is blocked by #692 

When we type something on the search input field, we have no way of deleting the content of the input field. We need a sort of cross button. Designs should be updated with instructions on when the cross button should show. Example from Google below.

<img width="726" alt="Screenshot 2023-12-04 at 15 50 03" src="https://github.com/international-labour-organization/designsystem/assets/22772102/4f73c42c-54bb-4042-bdd3-5c8f5464e900">

This is a low priority task to be addressed after all the other design tasks.

**Comment by beatrizmartinmartins** [comment]
This is done. For the X icon, we should only included it when it's filled. Check it [here](https://www.figma.com/file/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?type=design&node-id=587%3A4853&mode=design&t=bDY5mxoS52alNTX7-1)

**Comment by justintemps** [comment]
@inesdgomes why do we need this? Why can't the user just backspace?

**Comment by inesdgomes** [comment]
@justintemps As discussed, this improves the user experience on the search page and sorted index.

**Comment by Shashika6** [comment]
@justintemps Do we only do this one on twig or both twig/react ?

**Comment by Shashika6** [comment]
@justintemps I did the changes in react as well, Shall we do it in another PR as it will be easy to review as well.
